### ISSUE TITLE
Line Numbers

### ISSUE THREAD

**Issue by jdev082** [jdev082/editor#1]

**Comment by kccuber-scratch** [comment]
it looks like the div that holds the line number can be backspaced out of existence?

**Comment by jdev082** [comment]
oh? the one with content editing enabled. Then how would anyone type text then?

**Comment by imdevarsh** [comment]
OK, let me see. I think this would require some JS. also where is `script.js`?

**Comment by jdev082** [comment]
> OK, let me see. I think this would require some JS. also where is `script.js`?

i didnt push my changes yet.

**Comment by imdevarsh** [comment]
OK!

**Comment by jdev082** [comment]
I can't push my changes, i hate it when i forget to commit my changes before merging PRs, Now I can't merge it! I didn't add anything anyway.

**Comment by imdevarsh** [comment]
What happened? What's the error?

**Comment by jdev082** [comment]
@webdev03 i said i didnt even change much

**Comment by imdevarsh** [comment]
But why can't you push your changes?

**Comment by jdev082** [comment]
just fix the bug i cant fix :(

**Comment by imdevarsh** [comment]
https://stackoverflow.com/questions/1995370/adding-line-numbers-to-html-textarea

**Comment by jdev082** [comment]
[nvm]

**Comment by imdevarsh** [comment]
You could use a `textarea` if this works.

**Comment by jdev082** [comment]
@webdev03 im not using jquery

**Comment by jdev082** [comment]
I can use this cool copy paste code if i was boring 
```js
var editor = CodeMirror.fromTextArea(document.getElementById('code'), {
    lineNumbers: true,
    mode: 'text/x-perl',
    theme: 'abbott',
});

<script language="javascript" type="text/javascript" src="https://codemirror.net/lib/codemirror.js"></script>
<script language="javascript" type="text/javascript" src="https://codemirror.net/mode/perl/perl.js"></script>

<link rel="stylesheet" type="text/css" href="https://codemirror.net/lib/codemirror.css"></link>
<link rel="stylesheet" type="text/css" href="https://codemirror.net/theme/abbott.css"></link>

<textarea id="code" name="code">
if($cool_variable) {
    doTheCoolThing();     # it's PRETTY cool, imho
}</textarea>
```

**Comment by imdevarsh** [comment]
Ok, so https://codemirror.net/ may help. Or https://stackoverflow.com/a/61053609/17333186

**Comment by imdevarsh** [comment]
I'd suggest modifying https://stackoverflow.com/a/61053609/17333186 and it would work. But then we could also have another solution which I can tell you.

**Comment by jdev082** [comment]
code mirror doesn't have line numbers

**Comment by imdevarsh** [comment]
It does. But then it would ruin the whole point of making it ourself if we use it.

**Comment by jdev082** [comment]
@webdev03 No it doesn't, very clearly when I copied and pasted the code it didn't have anything but an editable text area.

**Comment by kccuber-scratch** [comment]
no codemirror, that's cheaty

**Comment by jdev082** [comment]
@kccuber-scratch with plain HTML/CSS/JS (which i want to do it in) that's the only option.

**Comment by jdev082** [comment]
https://www.thatsoftwaredude.com/content/8912/create-a-basic-text-editor-in-javascript
the tutorial i originally followed

**Comment by jdev082** [comment]
the site is unusable because of the unblockable popups you cant get rid of.

**Comment by kccuber-scratch** [comment]
> the site is unusable because of the unblockable popups you cant get rid of.

it literally has 2 popups, the other can be blocked by an adblocker

**Comment by jdev082** [comment]
> > the site is unusable because of the unblockable popups you cant get rid of.
> 
> it literally has 2 popups, the other can be blocked by an adblocker

both are in the WAY of the stuff on the web page.

**Comment by jdev082** [comment]
also the guide doesn't have line numbers, which i need

**Comment by kccuber-scratch** [comment]
> > > the site is unusable because of the unblockable popups you cant get rid of.
> > 
> > 
> > it literally has 2 popups, the other can be blocked by an adblocker
> 
> both are in the WAY of the stuff on the web page.

no, i was able to close them. on the first one theres a big text that says "X close" and on the second one there's a small "no thanks" at the bottom right corner

**Comment by jdev082** [comment]
there's no X on the email asking one.

**Comment by kccuber-scratch** [comment]
> there's no X on the email asking one.

it's right below the box that asks for your email, screenshot pls

**Comment by jdev082** [comment]
![image](https://user-images.githubusercontent.com/92550746/147709340-39183082-9b18-4667-959a-d199acd14e4b.png)

here's what i found (and pasted) so far.
i have found copied and pasted all the source. so this is what i have.

**Comment by jdev082** [comment]
still no line numbers tho.

**Comment by jdev082** [comment]
i cant type in it anymore?

**Comment by jdev082** [comment]
fixed.

**Comment by jdev082** [comment]
none of the formatting buttons work tho

**Comment by jdev082** [comment]
nvm they do.

**Comment by jdev082** [comment]
not on gh-pages tho, github messed up my changes. and so did vscode.

**Comment by jdev082** [comment]
fixed.

**Comment by imdevarsh** [comment]
???

**Comment by imdevarsh** [comment]
@JaydenDev did you switch to VSCode? I thought you used vim.

**Comment by jdev082** [comment]
@webdev03 I use vscode any time I'm dealing with multiple files, it also has better syntax highlighting. Also a built in terminal and ... [...]

**Comment by imdevarsh** [comment]
Oh, ok. I use it too, you don't need to explain why it's better.

**Comment by jdev082** [comment]
Also it's nice to use jefallo's live server hosting extension for VSCODE. so that I can easily have it automatically reload every file save and have an easy server hosted on localhost for me to test.

**Comment by imdevarsh** [comment]
Same! Jeffalo's extension is awesome for working on static projects 🤣

**Comment by jdev082** [comment]
![image](https://user-images.githubusercontent.com/92550746/147719230-84059155-0a3e-48fa-aa63-6144a758b7bb.png)

**Comment by jdev082** [comment]
gh copilot doesn't work since im still in the waiting list.

**Comment by imdevarsh** [comment]
I just joined the waiting list. I hope I get in! :smile:
